starknet/src/utils/types.cairo

```diff
@@ -534,3 +535,121 @@ impl IndexedStrategyImpl of IndexedStrategyTrait {
+    fn should_update(self: @T) -> bool;
+}
+
+// Obtained by keccak256 hashing the string "No update", and then taking the corresponding number of bytes.
+// Evaluates to: 0xf2cda9b13ed04e585461605c0d6e804933ca828111bd94d4e6a96c75e8b048ba
+
+impl NoUpdateU32 of NoUpdateTrait<u32> {
+    fn no_update() -> u32 {
+        0xf2cda9b1
+    }
+
+    fn should_update(self: @u32) -> bool {
+        *self != 0xf2cda9b1
+    }
+}
+
+impl NoUpdateU64 of NoUpdateTrait<u64> {
+    fn no_update() -> u64 {
+        0xf2cda9b13ed04e58
+    }
+
+    fn should_update(self: @u64) -> bool {
+        *self != 0xf2cda9b13ed04e58
+    }
+}
+
+impl NoUpdateFelt252 of NoUpdateTrait<felt252> {
+    fn no_update() -> felt252 {
+        // First 248 bits
+        0xf2cda9b13ed04e585461605c0d6e804933ca828111bd94d4e6a96c75e8b048
+    }
+
+    fn should_update(self: @felt252) -> bool {
+        *self != 0xf2cda9b13ed04e585461605c0d6e804933ca828111bd94d4e6a96c75e8b048
+    }
+}
+
+impl NoUpdateContractAddress of NoUpdateTrait<ContractAddress> {
+    fn no_update() -> ContractAddress {
+        // First 248 bits
+        contract_address_const::<0xf2cda9b13ed04e585461605c0d6e804933ca828111bd94d4e6a96c75e8b048>()
+    }
+
+    fn should_update(self: @ContractAddress) -> bool {
+        *self != contract_address_const::<0xf2cda9b13ed04e585461605c0d6e804933ca828111bd94d4e6a96c75e8b048>()
+    }
+}
+
+impl NoUpdateStrategy of NoUpdateTrait<Strategy> {
+    fn no_update() -> Strategy {
+        Strategy {
+            address: contract_address_const::<0xf2cda9b13ed04e585461605c0d6e804933ca828111bd94d4e6a96c75e8b048>(),
+            params: array::ArrayTrait::new(),
+        }
+    }
+
+    fn should_update(self: @Strategy) -> bool {
+        *self
+            .address != contract_address_const::<0xf2cda9b13ed04e585461605c0d6e804933ca828111bd94d4e6a96c75e8b048>()
+    }
+}
+
+// TODO: find a way for "Strings"
+impl NoUpdateArray<T> of NoUpdateTrait<Array<T>> {
+    fn no_update() -> Array<T> {
+        array::ArrayTrait::<T>::new()
+    }
+
+    fn should_update(self: @Array<T>) -> bool {
+        self.len() != 0
+    }
+}
+
+
+impl UpdateSettingsCalldataImpl of UpdateSettingsCalldataTrait {
+    fn default() -> UpdateSettingsCalldata {
+        UpdateSettingsCalldata {
+            min_voting_duration: NoUpdateU64::no_update(),
+            max_voting_duration: NoUpdateU64::no_update(),
+            voting_delay: NoUpdateU64::no_update(),
+            metadata_URI: NoUpdateArray::no_update(),
+            dao_URI: NoUpdateArray::no_update(),
+            proposal_validation_strategy: NoUpdateStrategy::no_update(),
+            proposal_validation_strategy_metadata_URI: NoUpdateArray::no_update(),
+            authenticators_to_add: NoUpdateArray::no_update(),
+            authenticators_to_remove: NoUpdateArray::no_update(),
+            voting_strategies_to_add: NoUpdateArray::no_update(),
+            voting_strategies_metadata_URIs_to_add: array::ArrayTrait::new(),
```

Comment:
why not use `NoUpdateArray::no_update()`?

Reply:
Yeah it was a miss; it's because at first I wanted to have something visually different to remind me that `NoUpdateString::no_update()` has not been implemented yet...

We can go with `empty` for now but I think we'll need to change in the future because `""` is a valid metadata_uri and I'd be happier with something like `No Update` as a string to indicate it

Reply:
so we can change to `NoUpdateArray::no_update()` ?

Reply:
aye! will do